### PR TITLE
[SPARK-48007][BUILD][TESTS] Upgrade `mssql.jdbc` to `12.6.1.jre11`

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR upgrades mssql.jdbc.version to 12.6.1.jre11, https://mvnrepository.com/artifact/com.microsoft.sqlserver/mssql-jdbc.

### Why are the changes needed?

test dependency management

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSQLServerDatabaseOnDocker.scala [hunk 1]
```diff
@@ -28,5 +28,6 @@ class MsSQLServerDatabaseOnDocker extends DatabaseOnDocker {
   override val jdbcPort: Int = 1433
 
   override def getJdbcUrl(ip: String, port: Int): String =
-    s"jdbc:sqlserver://$ip:$port;user=sa;password=Sapass123;"
+    s"jdbc:sqlserver://$ip:$port;user=sa;password=Sapass123;" +
+      "encrypt=true;trustServerCertificate=true"
```
Comment (dongjoon-hyun) [inline connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSQLServerDatabaseOnDocker.scala:32]: This is enforced by default from now?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSQLServerDatabaseOnDocker.scala (hunk 1 above)
Comment (yaooqinn) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/MsSQLServerDatabaseOnDocker.scala:32]: Yeah, I referred and confirmed with ms-jdbc docs, which says it's recommended for testing purposes. I didn't investigate thoroughly when this change for default values was applied due to the large version span.

**Comment by yaooqinn** [comment]
Thank you @dongjoon-hyun

**Pull request closed without merging**